### PR TITLE
Updating EKSIntegration for Enhanced Container Insights

### PR DESCRIPTION
**Pull request by mitali-salvi**
# Description of the issue
Capture enhanced container insights metrics for EKSIntegrationTest

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
https://github.com/aws/amazon-cloudwatch-agent/actions/runs/6685532283/job/18163971709

### CODE REVIEW

File: test/metric_value_benchmark/eks_resources/test_schemas/control_plane.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    "apiserver_storage_db_total_size_in_bytes": {},
+    "etcd_db_total_size_in_bytes": {},
+    "apiserver_storage_list_duration_seconds": {},
+    "apiserver_longrunning_requests": {},
+    "apiserver_request_duration_seconds": {},
+    "rest_client_request_duration_seconds": {},
+    "apiserver_request_total": {},
+    "apiserver_request_total_5xx": {},
+    "apiserver_admission_controller_admission_duration_seconds": {},
+    "apiserver_admission_step_admission_duration_seconds",
```
Comment (dchappa) [inline test/metric_value_benchmark/eks_resources/test_schemas/control_plane.json:22]: is this missing an `{}` ?

File: terraform/eks/daemon/default_resources/default_amazon_cloudwatch_agent.json [hunk 2]
```diff
@@ -2,7 +2,8 @@
   "logs": {
     "metrics_collected": {
       "kubernetes": {
-        "metrics_collection_interval": 30
+        "metrics_collection_interval": 30,
+        "enhanced_container_insights": true
```
Comment (dchappa) [inline terraform/eks/daemon/default_resources/default_amazon_cloudwatch_agent.json:6]: Not going to block -- would it be better to have a separate test for enhanced_container_insights ?

File: test/metric_value_benchmark/eks_resources/test_schemas/control_plane.json (hunk 1 above)
Comment (mitali-salvi) [reply test/metric_value_benchmark/eks_resources/test_schemas/control_plane.json]: Good catch, updated

File: terraform/eks/daemon/default_resources/default_amazon_cloudwatch_agent.json (hunk 2 above)
Comment (mitali-salvi) [reply terraform/eks/daemon/default_resources/default_amazon_cloudwatch_agent.json:6]: Will be addressed in a separate test which tests every observation

**Review by JayPolanco — APPROVED** [review]

**Review by dchappa — APPROVED** [review]

**Pull request merged**